polysemy-plugin/src/Polysemy/Plugin/Fundep/Unification.hs

```diff
@@ -72,7 +71,16 @@ unify solve_ctx = tryUnifyUnivarsButNotSkolems skolems
         InterpreterUse _ s -> s
         FunctionDef s      -> s
 
-
+#if __GLASGOW_HASKELL__ >= 902
+tryUnifyUnivarsButNotSkolems :: Set TyVar -> Type -> Type -> Maybe TCvSubst
+tryUnifyUnivarsButNotSkolems skolems goal inst =
+  case tcUnifyTysFG
+         (bool (const BindMe) (const Apart) . flip S.member skolems)
```

Comment:
This is the only bit that's different, no? For things like this I like to just define some CPP:

```haskell
#if __GLASGOW_HASKEL__ >= 902
#define BINDME (const BindMe)
#define APART (const Apart)
#else
#define BINDME BindMe
#define APART Skolem
#endif


tryUnifyUnivarsButNotSkolems :: Set TyVar -> Type -> Type -> Maybe TCvSubst
tryUnifyUnivarsButNotSkolems skolems goal inst =
  case tcUnifyTysFG
         (bool BINDME SKOLEM . flip S.member skolems)
         [inst]
         [goal] of
    Unifiable subst -> pure subst
    _               -> Nothing
```

Reply:
:heavy_check_mark: